src/jdk.jfr/share/classes/jdk/jfr/internal/TypeLibrary.java

```diff
@@ -98,6 +98,9 @@ private static ValueDescriptor createDurationField() {
     }
 
     public static synchronized void initialize() {
+        if (initialized) {
+            return;
+        }
```

Comment:
Perhaps it would make sense to add a comment as to why that check needs to be there? Something like this perhaps?

```
// TypeLibrary is initialized once by MetadataRepository (which is a singleton). So shouldn't happen practice.
// However, some tools (GraalVM native image) may call the initialization routine twice.
```

Reply:
Thanks!